idea/src/org/jetbrains/kotlin/idea/refactoring/kotlinRefactoringUtil.kt

```diff
@@ -749,7 +749,15 @@ fun <ListType : KtElement> replaceListPsiAndKeepDelimiters(
     val lastOriginalParameter = oldParameters.last()
 
     if (oldCount > commonCount) {
-        originalList.deleteChildRange(oldParameters[commonCount - 1].nextSibling, lastOriginalParameter)
+        oldParameters[commonCount - 1]
+            .siblings(forward = true, withItself = false)
+            .takeWhile { it !== lastOriginalParameter }
+            .firstOrNull { it.node.elementType == KtTokens.COMMA }
+            ?.delete()
+        originalList.deleteChildRange(
+            lastOriginalParameter.prevSibling?.takeIf { it is PsiWhileStatement } ?: lastOriginalParameter,
```

Comment:
Can you please explain or write a test why `PsiWhileStatement` is special here?

Reply:
This is for removing unnecessary spaces before the parameter, but I removed it because it was unnecessary.